libraries/core_libs/network/src/tarcap/packets_handlers/pbft_block_packet_handler.cpp

```diff
@@ -26,64 +29,62 @@ void PbftBlockPacketHandler::process(const PacketData &packet_data, const std::s
   LOG(log_tr_) << "In PbftBlockPacket";
 
   auto pbft_block = std::make_shared<PbftBlock>(packet_data.rlp_[0]);
-  const uint64_t peer_pbft_chain_size = packet_data.rlp_[1].toInt();
-  LOG(log_tr_) << "Receive proposed PBFT Block " << pbft_block->getBlockHash().abridged()
-               << ", Peer PBFT Chain size: " << peer_pbft_chain_size << " from " << peer->getId();
+  const auto proposed_block_hash = pbft_block->getBlockHash();
+  const auto proposed_period = pbft_block->getPeriod();
+  LOG(log_tr_) << "Receive proposed PBFT Block " << proposed_block_hash.abridged() << ", for proposed period "
+               << proposed_period << " from " << peer->getId();
 
   peer->markPbftBlockAsKnown(pbft_block->getBlockHash());
+
+  const auto peer_pbft_chain_size = proposed_period - 1;
   if (peer_pbft_chain_size > peer->pbft_chain_size_) {
     peer->pbft_chain_size_ = peer_pbft_chain_size;
   }
 
-  const auto pbft_synced_period = pbft_mgr_->pbftSyncingPeriod();
-  if (pbft_synced_period >= pbft_block->getPeriod()) {
-    LOG(log_tr_) << "Drop new PBFT block " << pbft_block->getBlockHash().abridged() << " at period "
-                 << pbft_block->getPeriod() << ", own PBFT chain has synced at period " << pbft_synced_period;
+  const auto pbft_chain_size = pbft_chain_->getPbftChainSize();
+  if (pbft_chain_size + 1 != proposed_period) {
+    LOG(log_tr_) << "Drop proposed PBFT block " << proposed_block_hash.abridged() << " at period " << proposed_period
+                 << ", own PBFT chain size is " << pbft_chain_size;
```

Comment:
is this ok? as before we where ok when blocks where not yet executed

Reply:
pbft_chain has blocks when block get finalized, not require block get executed.
If proposal blocks are more than 1 period to node own pbft blocks, node has to sync PBFT blocks and not be able to vote on the proposal blocks.

Reply:
@MatusKysel I hit some missing proposal PBFT blocks issue. I have to re-thing about this

Reply:
I will change it back

Reply:
Fixed